src/core/queue/DatabaseQueue.ts

```diff
@@ -0,0 +1,15 @@
+import BaseQueue, {IQueue} from './BaseQueue';
+
+class DatabaseQueue extends BaseQueue {
+  static dbInstance: DatabaseQueue;
+
+  static getInstance(): DatabaseQueue {
+    if (!this.dbInstance) {
+      return new DatabaseQueue();
+    }
+
+    return this.dbInstance;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The singleton pattern implementation in `DatabaseQueue` seems incorrect. The `getInstance()` method creates a new instance every time it's called when `dbInstance` is not defined, but it doesn't assign this new instance to `dbInstance`. This means that `dbInstance` will always be `undefined`, and a new `DatabaseQueue` will be created each time `getInstance()` is called, which defeats the purpose of the singleton pattern.

Here's how you can fix it:

```diff
  static getInstance(): DatabaseQueue {
    if (!this.dbInstance) {
-     return new DatabaseQueue();
+     this.dbInstance = new DatabaseQueue();
    }

    return this.dbInstance;
  }
```
This way, the first call to `getInstance()` will create a new `DatabaseQueue` and store it in `dbInstance`, and subsequent calls will return this stored instance.


<!-- This is an auto-generated comment by OSS CodeRabbit -->